src/index.ts

```diff
@@ -80,7 +80,23 @@ export type Auth =
 	| DatabaseAuth
 	| ScopeAuth;
 
-export default class Surreal extends Emitter {
+export default class Surreal extends Emitter<
+	& {
+		open: [];
+		opened: [];
+		close: [];
+		closed: [];
+		notify: [any];
+	}
+	& {
+		[
+			K in Exclude<
+				string,
+				"open" | "opened" | "close" | "closed" | "notify"
+			>
+		]: [Result<any>];
+	}
+> {
```

Comment:
@mathe42 you asked for a difference; this is a little redundant on a nit-picky level. The reason for this is the anonymous type as it can't be reused for the Exclude.

This is my approach:
https://github.com/surrealdb/surrealdb.js/pull/44/files#diff-a2a171449d862fe29692ce031981047d7ab755ae7f84c707aef80701b3ea0c80R83-R99

So for in your case, what do you think about:

```ts
interface Whatever {
 		open: [];
 		opened: [];
 		close: [];
 		closed: [];
 		notify: [any];
}

export default class Surreal extends Emitter<
 	& Whatever
 	& {
 		[
 			K in Exclude<
 				string,
 				keyof Whatever
 			>
 		]: [Result<any>];
 	}
 > {
 ```

Reply:
Yes agree I will change that.